packages/@react-spectrum/searchwithin/test/SearchWithin.test.js

```diff
@@ -0,0 +1,109 @@
+import {theme} from '@react-spectrum/theme-default';
+import {triggerPress} from '@react-spectrum/test-utils';
+import {typeText} from '@react-spectrum/test-utils';
+
+let defaultProps = {
+  label: 'Test'
+};
+
+function renderSearchWithin(props = {}, searchFieldProps = {}, pickerProps = {}) {
+  return render(
+    <Provider theme={theme}>
+      <SearchWithin {...defaultProps} {...props}>
+        <SearchField placeholder="Search" {...searchFieldProps} />
+        <Picker defaultSelectedKey="all" {...pickerProps}>
+          <Item key="all">All</Item>
+          <Item key="campaigns">Campaigns</Item>
+          <Item key="audiences">Audiences</Item>
+          <Item key="tags">Tags</Item>
+        </Picker>
+      </SearchWithin>
+    </Provider>  
+  );
+}
+
+describe('SearchWithin', function () {
+  beforeAll(function () {
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    act(() => jest.runAllTimers());
+  });
+
+  afterAll(function () {
+    jest.restoreAllMocks();
+  });
+
+  it('renders correctly', function () {
+    let {getAllByText, getByRole, getByLabelText} = renderSearchWithin();
+
+    let searchfield = getByLabelText('Test', {selector: 'input'});
+    expect(searchfield).toBeVisible();
+    expect(searchfield).toHaveAttribute('type', 'search');
+
+    let button = getByRole('button');
+    expect(button).toBeVisible();
+    expect(button).toHaveAttribute('aria-haspopup', 'listbox');
+
+    let label = getAllByText('Test')[0];
+    expect(label).toBeVisible();
+  });
+
+  it('can type in search and get onChange', function () {
+    let onChange = jest.fn();
+    let {getByLabelText} = renderSearchWithin({}, {onChange});
+    let searchfield = getByLabelText('Test', {selector: 'input'});
+    expect(searchfield).toHaveAttribute('value', '');
+
+    act(() => {searchfield.focus();});
+    typeText(searchfield, 'test search');
+    act(() => {searchfield.blur();});
+    expect(searchfield).toHaveAttribute('value', 'test search');
+    expect(onChange).toBeCalledTimes(11);
+  });
+
+  it('can open menu and get onChange', function () {
+    let onOpenChange = jest.fn();
+    let {getByRole} = renderSearchWithin({}, {}, {onOpenChange});
+
+    let picker = getByRole('button');
+    triggerPress(picker);
+
+    let listbox = getByRole('listbox');
+    expect(listbox).toBeVisible();
+    expect(onOpenChange).toBeCalledTimes(1);
+    expect(onOpenChange).toHaveBeenCalledWith(true);
+  });
+
+  it('searchfield and picker are labelled correctly', function () {
+    let {getByRole, getAllByText, getByLabelText} = renderSearchWithin();
+
+    let searchfield = getByLabelText('Test', {selector: 'input'});
+    let picker = getByRole('button');
+    triggerPress(picker);
+
+    let listbox = getByRole('listbox');
+    let label = getAllByText('Test')[0];
+    expect(listbox).toHaveAttribute('aria-labelledby', label.id);
+    expect(searchfield).toHaveAttribute('aria-labelledby', label.id);
+  });
```

Comment:
Would be nice to have tests for `isDisabled`, `autoFocus`, one for making sure that the group element is has the expected aria-labelledby value (prob just need to add another line to the test above checking that), that `label`/`isQuiet`/`isRequired` provided to child SearchField/Picker are properly overridden by the SearchWithin slot props